docs/api.md

```diff
@@ -70,6 +70,7 @@ These are the API group versions which are curently available when using Antrea.
 | CRD | CRD version | Introduced in | Deprecated in | Removed in |
 |---|---|---|---|---|
 | `ClusterGroup` | v1alpha2 | v1.0.0 | v1.1.0 | v1.12.0 [^1] |
+| `Tier` | v1alpha1 | v1.0.0 | v1.13.0 | v2.0.0 |
```

Comment:
@antoninbas could this lead to confusion when reading it? and it seems making the `Planned Removal` column in the "Currently-supported" section meaningless if we directly move deprecated but supported version here.

Reply:
Yes, I missed that. It should stay in the previous section until it is actually removed.

Reply:
Cool. Changed.